crates/holochain_integrity_types/src/entry.rs

```diff
@@ -45,17 +47,26 @@ impl From<EntryHashed> for Entry {
 }
 
 /// Structure holding the entry portion of a chain record.
+pub type Entry = EntryImpl<AppEntryBytes>;
+
+/// Data for building a [`Record`](crate::record::Record).
+pub type RecordBuilder = EntryImpl<AppEntry>;
```

Comment:
What do you think about calling this something else? My objections is that this doesn't really evoke [the Builder pattern](https://rust-unofficial.github.io/patterns/patterns/creational/builder.html). `build()` doesn't even produce the thing that this is supposed to be a Builder for (a Record), it just produces some intermediate state, which has to be built up into a record elsewhere.

```suggestion
/// All entry-specific data needed to build a [`Record`](crate::record::Record).
pub type EntryPlus = EntryImpl<AppEntry>;
```